metaflow/plugins/argo/argo_inner_flow.py

```diff
@@ -34,7 +67,104 @@ def __init__(
         self._argo_run_id = self._flow_information["metadata"]["name"]
         self._metaflow_run_id = f"argo-{self._argo_run_id}"
         self._kubernetes_namespace = self._flow_information["metadata"]["namespace"]
-        self._exception = None
-        self._status = None
 
-        print(f"Inner flow has started w/ Argo id: {self._argo_run_id}")
+        # Waiting for Argo workflow to trigger is not optional.
+        # It is necessary to determine Flow name if not given as parameter
+        wait_to_trigger = 20  # wait time is 20 seconds
+        print("attempting to trigger Argo workflow")
+        while wait_to_trigger > 0 and self.status is None:
+            time.sleep(1)
+            wait_to_trigger -= 1
+
+        if wait_to_trigger == 0:
+            raise Exception("Inner flow failed to begin running")
+
+        print(
+            f"A run of Metaflow flow {self._flow_name} has started w/ Argo id: {self._argo_run_id}"
+        )
+
+        # [optional] wait for argo workflow to finish running
+        if self._wait:
+            print("\nNow we will wait for the flow to finish")
+            wait_remaining = self._wait_to_run  # in minutes (5s for testing)
+            print(f"status: {self.status}, 'mins' to wait: {wait_remaining}")
+            while wait_remaining > 0 and self.status == "Running":
+                wait_remaining -= 1
+                print(f"status: {self.status}, 'mins' to wait: {wait_remaining}")
+                time.sleep(5)
+
+            if wait_remaining == 0:
+                print("Inner flow timed out. better work on that speed for next time!")
+
+            print("\nInner flow is finished!!!")
+
+        else:
+            print("Not waiting for inner flow to finish")
+
+            
+    @property
+    def status(self):
+        wf = self._argo_client.get_workflow(self._argo_run_id)
+
+        if "status" in wf and "phase" in wf["status"]:
+            self._status = wf["status"]["phase"]
```

Comment:
`wf.get("status").get("phase")`

`get()` help you avoid checking if a key exists in dictionary.
Reference: https://docs.python.org/3/library/stdtypes.html#dict.get

Reply:
Well I think your code is ok - with my suggestion value of `wf.get("status")` still needs to be checked, unless status is always presented - which I do not know for sure.

Reply:
I got an error when I used 'wf.get("status").get("phase")' in the case where "status" was not found because:

wf.get("status") becomes None, and then I get TypeError: argument of type 'NoneType' is not iterable

Solution: I check 'if workflow.get("status")' and then try to find if "phase" is a key.